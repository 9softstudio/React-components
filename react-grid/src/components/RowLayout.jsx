import React, { Component } from 'react';
import PropTypes from 'prop-types';

class RowLayout extends Component {
    constructor(props) {
        super(props)
    }

    static propTypes = {
        columnLayout: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.number, PropTypes.string])).isRequired
    }

    _getRowLayout() {
        const { columnLayout } = this.props;

        return columnLayout.map((cellWidth, index) => {
            const width = isNaN(cellWidth) ? cellWidth : `${cellWidth}px`;

            return (<td key={`cellLayout${index}`} style={{ width: width }}></td>);
        });
    }

    render() {
        return (
            <tr className="first-row">
                {this._getRowLayout()}
            </tr>
        );
    }
}

export default RowLayout;