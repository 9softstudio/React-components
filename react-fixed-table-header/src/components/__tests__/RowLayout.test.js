import React from 'react'
import RowLayout from '../RowLayout'
import renderer from 'react-test-renderer'

describe('RowLayout ', () => {
    afterAll(() => {
        jest.resetAllMocks();
    })

    describe('_getRowLayout', () => {
        it('columnLayout is array of number - props of children should be correct', () => {
            const props = {
                columnLayout: [100, 200]
            }

            const rowLayout = new RowLayout(props);
            const renderedRowLayout = rowLayout._getRowLayout();

            renderedRowLayout.forEach((child, index) => {
                const expectedKey = `cellLayout${index}`;
                const width = props.columnLayout[index];
                const expectedStyle = { width: `${width}px` };

                const keyResult = child.key;
                const styleResult = child.props.style;

                expect(keyResult).toBe(expectedKey);
                expect(styleResult).toEqual(expectedStyle);
            })
        })

        it('columnLayout is array of string - props of children should be correct', () => {
            const props = {
                columnLayout: ['20%', '30%']
            }

            const rowLayout = new RowLayout(props);
            const renderedRowLayout = rowLayout._getRowLayout();

            renderedRowLayout.forEach((child, index) => {
                const expectedKey = `cellLayout${index}`;
                const width = props.columnLayout[index];
                const expectedStyle = { width };

                const keyResult = child.key;
                const styleResult = child.props.style;

                expect(keyResult).toBe(expectedKey);
                expect(styleResult).toEqual(expectedStyle);
            })
        })
    })

    it('rendered content should be match with snapshot', () => {
        const props = {
            columnLayout: [100, 200]
        }

        const rowLayout = renderer.create(<RowLayout {...props} />);

        const result = rowLayout.toJSON();

        expect(result).toMatchSnapshot();
    })
})