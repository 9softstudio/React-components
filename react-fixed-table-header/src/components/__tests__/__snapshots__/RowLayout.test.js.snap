// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowLayout  rendered content should be match with snapshot 1`] = `
<tr
  className="first-row"
>
  <td
    style={
      Object {
        "width": "100px",
      }
    }
  />
  <td
    style={
      Object {
        "width": "200px",
      }
    }
  />
</tr>
`;
